lib/checks/report/report.rb

```diff
@@ -0,0 +1,28 @@
+module Checks
+  module Report
+    class Report
```

Comment:
The purpose of the report is to encapsulate the stuff the checks need to return.
We could certainly move the whitelisting out, but not the report creation.